app/apptesting/test_helpers.go

```diff
@@ -261,3 +262,21 @@ func CopyConnectionAndClientToPath(path *ibctesting.Path, pathToCopy *ibctesting
 	path.EndpointB.ConnectionConfig = pathToCopy.EndpointB.ConnectionConfig
 	return path
 }
+
+func  (s *AppTestHelper) ICAPacketAcknowledgement(msgs []sdk.Msg) channeltypes.Acknowledgement {
+	txMsgData := &sdk.TxMsgData{
+		Data: make([]*sdk.MsgData, len(msgs)),
+	}
+	for i, msg := range msgs {
+		msgResponse := []byte("msg_response")
+		txMsgData.Data[i] = &sdk.MsgData{
+			MsgType: sdk.MsgTypeURL(msg),
+			Data:    msgResponse,
+		}
+
+	}
+	marshalledTxMsgData, err := proto.Marshal(txMsgData)
```

Comment:
should use use the codec marshaller instead of proto here?

Reply:
I mimicked the ibc module here https://github.com/cosmos/ibc-go/blob/8306273a5b1dbf828e4af5249896b1cfa2da5906/modules/apps/27-interchain-accounts/host/keeper/relay.go#L83